pkg/tunnel/util_test.go

```diff
@@ -0,0 +1,324 @@
+package tunnel
+
+import (
+	"fmt"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+	"reflect"
+	"testing"
+)
+
+/**
+GenerateKeyPairWithString should generate Key with Nodename
+*/
+func TestGenerateKeyPairWithString(t *testing.T) {
+	//Benchmarks to identify different node names, including numbers and letters
+	cases := []struct {
+		given, wanted string
+	}{
+		{"k8s-master", "12D3KooWB5qVCMrMNLpBDfMu6o4dy6ci2UqDVsFVomcd2PfYVzfW"},
+		{"k8s-node1", "12D3KooWErZ4m27CEinjcnXNvem2KFUJUFRBcd9NdcWnYRqyr1Sn"},
+		{"ke-edge1", "12D3KooWF3RB8SoMRZht7MDqF6GoipSPdFxVb9ESRLU7NuPX6ifH"},
+		{"ke-edge2", "12D3KooWSD4f5fZb5c9PQ6FPVd8Em4eKX3mRezcyqXSHUyomoy8S"},
+		//other form of NodeName
+		{"k8s-Master", "12D3KooWB5qVCMrMNLpBDfMu6o4dy6ci2UqDVsFVomcd2PfYVzfW"},
+		{"mａsteｒ", "12D3KooWB5qVCMrMNLpBDfMu6o4dy6ci2UqDVsFVomcd2PfYVzfW"},
+		{"ke-node1\n", "12D3KooWErZ4m27CEinjcnXNvem2KFUJUFRBcd9NdcWnYRqyr1Sn"},
+	}
+	for _, c := range cases {
+		t.Run(c.given, func(t *testing.T) {
+			key, err := GenerateKeyPairWithString(c.given)
+			assertEqual(err, nil, t)
+			out, err := peer.IDFromPrivateKey(key)
+			assertEqual(err, nil, t)
+			if ans := peer.Encode(out); !assertEqual(ans, c.wanted, t) {
+				t.Fatalf("Node name is %s and expected key-id is %s, but %s got",
+					c.given, c.wanted, ans)
+			}
+		})
+	}
+	// other situation to deal with
+
+	cases2 := []struct {
+		given, wanted string
+	}{
+		//A null condition occurs,the answer is for null ,there is no err fead back
+		{"", ""},
+		//illegal IP address
+	}
+	for _, c := range cases2 {
+		t.Run(c.given, func(t *testing.T) {
+			key, err := GenerateKeyPairWithString(c.given)
+			if assertEqual(err, nil, t) {
+				out, err := peer.IDFromPrivateKey(key)
+				assertEqual(err, nil, t)
+				t.Fatalf("Node name is nil but still got Key-ID: %s",
+					peer.Encode(out))
+			}
+		})
+	}
+
+}
+
+/**
```

Comment:
/**/ is not the structure annotation method recommended by go, please replace it with //